src/printer/tvmscript_unified_printer.cc

```diff
@@ -0,0 +1,738 @@
+    return output_;
+  }
+
+  OutputStream output_;
+  int indent_ = 0;
+};
+
+class PythonDocPrinter : public DocPrinter {
+ public:
+  PythonDocPrinter(String tir_prefix) : tir_prefix_(tir_prefix) {}
+
+ protected:
+  using DocPrinter::PrintDoc;
+
+  virtual void PrintDoc(const LiteralValueDoc& doc) override {
+    auto& value = doc->value;
+    if (value->IsInstance<FloatImmNode>() || value->IsInstance<IntImmNode>()) {
+      PrintNumberNode(Downcast<PrimExpr>(doc->value));
+    } else if (value->IsInstance<StringObj>()) {
+      PrintStringLiteral(Downcast<String>(value));
+    } else if (value->IsInstance<tir::StringImmNode>()) {
+      PrintStringLiteral(Downcast<StringImm>(value)->value);
+    }
+  };
+
+  virtual void PrintDoc(const ConstDoc& doc) override {
+    switch (doc->kind) {
+      case ConstDocNode::ConstKind::TIRBuilder:
+        output_ << tir_prefix_;
+        break;
+      case ConstDocNode::ConstKind::RelaxBuilder:
+        LOG(FATAL) << "RelaxBuilder constant not supported yet";
+        break;
+      case ConstDocNode::ConstKind::None:
+        output_ << "None";
+        break;
+    }
+  };
+
+  virtual void PrintDoc(const IdentifierDoc& doc) override { output_ << doc->name; };
+
+  virtual void PrintDoc(const AttrAccessDoc& doc) override {
+    PrintDoc(doc->value);
+    output_ << ".";
+    PrintDoc(doc->attr);
+  };
+
+  virtual void PrintDoc(const IndexDoc& doc) override {
+    PrintDoc(doc->value);
+    PrintJoinedElements("[", doc->indices, ", ", "]", "()");
+  };
+
+  virtual void PrintDoc(const OperationDoc& doc) override {
+    PrintDoc(doc->operands[0]);
+    switch (doc->kind) {
+      case OperationDocNode::OperationKind::Add:
+        output_ << " + ";
+        break;
+      case OperationDocNode::OperationKind::Sub:
+        output_ << " - ";
+        break;
+      case OperationDocNode::OperationKind::Mul:
+        output_ << " * ";
+        break;
+      case OperationDocNode::OperationKind::Div:
+        output_ << " / ";
+        break;
+      case OperationDocNode::OperationKind::FloorDiv:
+        output_ << " // ";
+        break;
+    }
+    PrintDoc(doc->operands[1]);
+  };
+
+  virtual void PrintDoc(const CallDoc& doc) override {
+    PrintDoc(doc->callee);
+    PrintJoinedElements("(", doc->args, ", ", ")");
+  };
+
+  virtual void PrintDoc(const TupleDoc& doc) override {
+    auto size = doc->elements.size();
+    if (size == 0) {
+      output_ << "tuple()";
+    } else if (size == 1) {
+      output_ << "(";
+      PrintDoc(doc->elements[0]);
+      output_ << ",)";
+    } else {
+      PrintJoinedElements("(", doc->elements, ", ", ")");
+    }
+  };
+
+  virtual void PrintDoc(const SeqStmtDoc& doc) override {
+    for (auto& stmt : doc->seq) {
+      PrintDoc(stmt);
+    }
+  };
+
+  virtual void PrintDoc(const ScopeDoc& doc) override {
+    NewLine() << "with ";
+    PrintDoc(doc->scope);
+    output_ << ":";
+
+    PrintWithIncreasedIndent(doc->body);
```

Comment:
probably need to improve the naming a little bit, but i dont have specific idea

```suggestion
    PrintIndentScope(doc->body);
```